contracts/EthProxy.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.6.7;
+
+import "@openzeppelin/contracts/utils/Address.sol";
+import "./IUSM.sol";
+import "./external/IWETH9.sol";
+
+
+contract EthProxy {
+    using Address for address payable;
+    IUSM public usm;
+    IWETH9 public weth;
+
+    constructor(address usm_, address weth_) public {
+        usm = IUSM(usm_); 
+        weth = IWETH9(weth_);
+
+        weth.approve(address(usm), uint(-1));
+    }
+
+    /// @dev The WETH9 contract will send ether to EthProxy on `weth.withdraw` using this function.
+    receive() external payable { }
+
+    /// @dev Users use `mint` in EthProxy to post ETH to USM (amount = msg.value), which will be converted to Weth here.
+    /// @param to Receiver of the minted USM
+    function mint(address to)
+        external payable {
+        weth.deposit{ value: msg.value }();
+        usm.mint(address(this), to, msg.value);
+    }
+
+    /// @dev Users wishing to withdraw their Weth as ETH from USM should use this function.
+    /// Users must have called `controller.addDelegate(ethProxy.address)` to authorize EthProxy to act in their behalf.
+    /// @param to Receiver of the obtained Eth
+    /// @param usmToBurn Amount of USM to burn.
+    function burn(address payable to, uint usmToBurn)
+        external {
+        uint wethToWithdraw = usm.burn(msg.sender, address(this), usmToBurn);
+        weth.withdraw(wethToWithdraw);
+        to.sendValue(wethToWithdraw);
+    }
```

Comment:
Same for consistency of the return value here, and for the other fns...

Reply:
Fixed